### ISSUE TITLE
authorName doesn't work

### ISSUE THREAD

**Issue by zoidberg8e** [lsowen/ep_auth_author#2]
Hello,
the authorName setting doesnt do anything at all,
Users still have no author name.

It also doesn't work without the authorName

Here is an example:

  "ep_auth_author": {
    "prefix": "author/"
  },  
  "users": {
    "author/test": {
      "authorName": "TestUser",
      "password": "changeme",
      "is_admin": false
    },  
  }

This leads to the user being unnamed.
which is the same if i do :

  "ep_auth_author": {
    "prefix": "author/"
  },  
  "users": {
    "author/test": {
//      "authorName": "TestUser",
      "password": "changeme",
      "is_admin": false
    },  
  }

Thank you

**Comment by lsowen** [comment]
A few questions to help diagnose:
1. What version of elasticpad do you have?
2. Are you logging in with the username `test` or `author/test`?
3. In settings/Troubleshooting Information (`admin/plugins/info`), do you see the following line?

```
     authenticate
         ep_auth_author/ep_auth_author from ep_auth_author/ep_auth_author
```

**Comment by zoidberg8e** [comment]
Thanks for the quick reply.
I use etherpad lite
 "Your Etherpad version is 1.5.6"
i login with username test
yes i see
authenticate
    ep_auth_author/ep_auth_author from ep_auth_author/ep_auth_author

**Comment by froiloc** [comment]
Hi, I'm facing the same issue.

troubleshooting information tells me: (\* more text after quote :) )

"
Etherpad version

Version number: 1.5.6

Git sha: 85d44c8

Installed plugins

ep_adminpads
ep_autoscrolldown, ep_auth_author, ep_colors, ep_copy_paste_select_all, ep_cursortrace, ep_copy_paste_images, ep_draw, ep_authornames, ep_embedmedia, ep_delete_empty_pads, ep_export_authors, ep_email_notifications, ep_export_background, ep_etherpad-lite, ep_export_documents_images, ep_hash_auth, ep_ldapauth, ep_headings, ep_mammoth, ep_mediawiki, ep_printer, ep_public_view, ep_scrollto, ep_spellcheck, ep_tables2, ep_horizontal_line, ep_scrolltoanchor, ep_themes
Installed parts

ep_themes/ep_themes
ep_scrolltoanchor/scrolltoanchor
ep_tables2/tables
ep_spellcheck/ep_spellcheck
ep_scrollto/scrollto
ep_public_view/ep_public_view
ep_printer/main
ep_mediawiki/mediawiki
ep_mammoth/ep_mammoth
ep_headings/main
ep_ldapauth/ep_ldapauth
ep_hash_auth/ep_hash_auth
ep_etherpad-lite/swagger
ep_etherpad-lite/adminsettings
ep_etherpad-lite/adminplugins
ep_etherpad-lite/admin
ep_etherpad-lite/tests
ep_etherpad-lite/socketio
ep_etherpad-lite/errorhandling
ep_etherpad-lite/importexport
ep_etherpad-lite/apicalls
ep_etherpad-lite/webaccess
ep_horizontal_line/horizontal_line
ep_export_documents_images/export_documents_images
ep_etherpad-lite/padreadonly
ep_etherpad-lite/padurlsanitize
ep_etherpad-lite/specialpages
ep_etherpad-lite/i18n
ep_etherpad-lite/static
ep_etherpad-lite/express
ep_export_background/main
ep_email_notifications/ep_email_notifications
ep_export_authors/ep_export_authors
ep_delete_empty_pads/ep_delete_empty_pads
ep_embedmedia/embedmedia
ep_authornames/authornames
ep_draw/draw
ep_copy_paste_images/ep_copy_paste_images
ep_cursortrace/cursortrace
ep_copy_paste_select_all/copy_paste_select_all
ep_colors/main
ep_auth_author/ep_auth_author
ep_autoscrolldown/autoscrolldown
ep_adminpads/ep_adminpads
Installed hooks

Server side hooks

clientVars
ep_themes/index:clientVars from ep_themes/ep_themes
clientVars
ep_email_notifications/client:clientVars from ep_email_notifications/ep_email_notifications
clientVars
ep_draw/draw:clientVars from ep_draw/draw
eejsBlock_scripts
ep_themes/index:eejsBlock_scripts from ep_themes/ep_themes
eejsBlock_scripts
ep_tables2/static/js/datatables:eejsBlock_scripts from ep_tables2/tables
eejsBlock_scripts
ep_mediawiki/index from ep_mediawiki/mediawiki
eejsBlock_scripts
ep_export_documents_images/hooks from ep_export_documents_images/export_documents_images
eejsBlock_scripts
ep_email_notifications/client from ep_email_notifications/ep_email_notifications
eejsBlock_scripts
ep_embedmedia/hooks:eejsBlock_scripts from ep_embedmedia/embedmedia
eejsBlock_scripts
ep_autoscrolldown/hooks from ep_autoscrolldown/autoscrolldown
eejsBlock_mySettings.dropdowns
ep_themes/index:eejsBlock_mySettings_dropdowns from ep_themes/ep_themes
eejsBlock_styles
ep_tables2/static/js/datatables:eejsBlock_styles from ep_tables2/tables
eejsBlock_styles
ep_printer/static/js/index.js from ep_printer/main
eejsBlock_styles
ep_horizontal_line/horizontalLine from ep_horizontal_line/horizontal_line
eejsBlock_styles
ep_export_documents_images/hooks from ep_export_documents_images/export_documents_images
eejsBlock_styles
ep_email_notifications/client:eejsBlock_styles from ep_email_notifications/ep_email_notifications
eejsBlock_styles
ep_embedmedia/hooks:eejsBlock_styles from ep_embedmedia/embedmedia
eejsBlock_styles
ep_copy_paste_images/index from ep_copy_paste_images/ep_copy_paste_images
eejsBlock_styles
ep_autoscrolldown/hooks from ep_autoscrolldown/autoscrolldown
eejsBlock_editbarMenuLeft
ep_tables2/static/js/datatables:eejsBlock_editbarMenuLeft from ep_tables2/tables
eejsBlock_editbarMenuLeft
ep_headings/index from ep_headings/main
eejsBlock_editbarMenuLeft
ep_horizontal_line/horizontalLine from ep_horizontal_line/horizontal_line
eejsBlock_editbarMenuLeft
ep_embedmedia/hooks:eejsBlock_editbarMenuLeft from ep_embedmedia/embedmedia
eejsBlock_editbarMenuLeft
ep_draw/draw from ep_draw/draw
eejsBlock_editbarMenuLeft
ep_colors/index from ep_colors/main
eejsBlock_editbarMenuLeft
ep_autoscrolldown/hooks from ep_autoscrolldown/autoscrolldown
eejsBlock_timesliderScripts
ep_tables2/static/js/datatables:eejsBlock_timesliderScripts from ep_tables2/tables
getLineHTMLForExport
ep_tables2/export2html:getLineHTMLForExport from ep_tables2/tables
getLineHTMLForExport
ep_mediawiki/export from ep_mediawiki/mediawiki
getLineHTMLForExport
ep_headings/index from ep_headings/main
getLineHTMLForExport
ep_horizontal_line/horizontalLine from ep_horizontal_line/horizontal_line
getLineHTMLForExport
ep_export_documents_images/hooks from ep_export_documents_images/export_documents_images
getLineHTMLForExport
ep_export_authors/index from ep_export_authors/ep_export_authors
getLineHTMLForExport
ep_colors/index from ep_colors/main
eejsBlock_mySettings
ep_spellcheck/spellcheck from ep_spellcheck/ep_spellcheck
eejsBlock_mySettings
ep_mediawiki/index from ep_mediawiki/mediawiki
eejsBlock_mySettings
ep_email_notifications/client:eejsBlock_mySettings from ep_email_notifications/ep_email_notifications
eejsBlock_dd_view
ep_spellcheck/spellcheck from ep_spellcheck/ep_spellcheck
expressConfigure
ep_public_view/index:expressConfigure from ep_public_view/ep_public_view
expressConfigure
ep_etherpad-lite/node/hooks/express/webaccess:expressConfigure from ep_etherpad-lite/webaccess
expressCreateServer
ep_public_view/index:expressServer from ep_public_view/ep_public_view
expressCreateServer
ep_mediawiki/express from ep_mediawiki/mediawiki
expressCreateServer
ep_etherpad-lite/node/hooks/express/swagger:expressCreateServer from ep_etherpad-lite/swagger
expressCreateServer
ep_etherpad-lite/node/hooks/express/adminsettings:expressCreateServer from ep_etherpad-lite/adminsettings
expressCreateServer
ep_etherpad-lite/node/hooks/express/adminplugins:expressCreateServer from ep_etherpad-lite/adminplugins
expressCreateServer
ep_etherpad-lite/node/hooks/express/admin:expressCreateServer from ep_etherpad-lite/admin
expressCreateServer
ep_etherpad-lite/node/hooks/express/tests:expressCreateServer from ep_etherpad-lite/tests
expressCreateServer
ep_etherpad-lite/node/hooks/express/socketio:expressCreateServer from ep_etherpad-lite/socketio
expressCreateServer
ep_etherpad-lite/node/hooks/express/errorhandling:expressCreateServer from ep_etherpad-lite/errorhandling
expressCreateServer
ep_etherpad-lite/node/hooks/express/importexport:expressCreateServer from ep_etherpad-lite/importexport
expressCreateServer
ep_etherpad-lite/node/hooks/express/apicalls:expressCreateServer from ep_etherpad-lite/apicalls
expressCreateServer
ep_export_documents_images/hooks from ep_export_documents_images/export_documents_images
expressCreateServer
ep_etherpad-lite/node/hooks/express/padreadonly:expressCreateServer from ep_etherpad-lite/padreadonly
expressCreateServer
ep_etherpad-lite/node/hooks/express/padurlsanitize:expressCreateServer from ep_etherpad-lite/padurlsanitize
expressCreateServer
ep_etherpad-lite/node/hooks/express/specialpages:expressCreateServer from ep_etherpad-lite/specialpages
expressCreateServer
ep_etherpad-lite/node/hooks/i18n:expressCreateServer from ep_etherpad-lite/i18n
expressCreateServer
ep_etherpad-lite/node/hooks/express/static:expressCreateServer from ep_etherpad-lite/static
expressCreateServer
ep_email_notifications/index:registerRoute from ep_email_notifications/ep_email_notifications
expressCreateServer
ep_delete_empty_pads/index:deletePadsAtStart from ep_delete_empty_pads/ep_delete_empty_pads
expressCreateServer
ep_adminpads/index:registerRoute from ep_adminpads/ep_adminpads
eejsBlock_editbarMenuRight
ep_printer/static/js/index.js from ep_printer/main
eejsBlock_exportColumn
ep_mediawiki/index from ep_mediawiki/mediawiki
eejsBlock_exportColumn
ep_export_documents_images/hooks from ep_export_documents_images/export_documents_images
import
ep_mammoth/import.js from ep_mammoth/ep_mammoth
collectContentPre
ep_headings/static/js/shared from ep_headings/main
collectContentPre
ep_horizontal_line/static/js/shared from ep_horizontal_line/horizontal_line
collectContentPre
ep_copy_paste_images/static/js/index from ep_copy_paste_images/ep_copy_paste_images
collectContentPre
ep_colors/static/js/shared from ep_colors/main
collectContentPost
ep_headings/static/js/shared from ep_headings/main
collectContentPost
ep_horizontal_line/static/js/shared from ep_horizontal_line/horizontal_line
collectContentPost
ep_copy_paste_images/static/js/index from ep_copy_paste_images/ep_copy_paste_images
collectContentPost
ep_colors/static/js/shared from ep_colors/main
authenticate
ep_ldapauth/ep_ldapauth from ep_ldapauth/ep_ldapauth
authenticate
ep_hash_auth/ep_hash_auth from ep_hash_auth/ep_hash_auth
authenticate
ep_auth_author/ep_auth_author from ep_auth_author/ep_auth_author
authorize
ep_ldapauth/ep_ldapauth from ep_ldapauth/ep_ldapauth
handleMessage
ep_ldapauth/ep_ldapauth from ep_ldapauth/ep_ldapauth
handleMessage
ep_email_notifications/handleMessage from ep_email_notifications/ep_email_notifications
handleMessage
ep_cursortrace/handleMessage from ep_cursortrace/cursortrace
handleMessage
ep_auth_author/ep_auth_author from ep_auth_author/ep_auth_author
socketio
ep_etherpad-lite/node/hooks/express/adminsettings:socketio from ep_etherpad-lite/adminsettings
socketio
ep_etherpad-lite/node/hooks/express/adminplugins:socketio from ep_etherpad-lite/adminplugins
socketio
ep_adminpads/index:socketio from ep_adminpads/ep_adminpads
createServer
ep_etherpad-lite/node/hooks/express:createServer from ep_etherpad-lite/express
restartServer
ep_etherpad-lite/node/hooks/express:restartServer from ep_etherpad-lite/express
stylesForExport
ep_export_background/index from ep_export_background/main
padUpdate
ep_email_notifications/update from ep_email_notifications/ep_email_notifications
userLeave
ep_delete_empty_pads/index:deletePadAtLeave from ep_delete_empty_pads/ep_delete_empty_pads
eejsBlock_body
ep_embedmedia/hooks:eejsBlock_body from ep_embedmedia/embedmedia
eejsBlock_timesliderStyles
ep_copy_paste_images/index from ep_copy_paste_images/ep_copy_paste_images
eejsBlock_dd_edit
ep_copy_paste_select_all/copy_paste_select_all from ep_copy_paste_select_all/copy_paste_select_all
padRemove
ep_adminpads/index:updatePads from ep_adminpads/ep_adminpads
padCreate
ep_adminpads/index:updatePads from ep_adminpads/ep_adminpads
eejsBlock_adminMenu
ep_adminpads/index:eejsBlock_adminMenu from ep_adminpads/ep_adminpads
Client side hooks

aceInitialized
ep_themes/static/js/init from ep_themes/ep_themes
aceInitialized
ep_tables2/static/js/datatables:aceInitialized from ep_tables2/tables
aceInitialized
ep_copy_paste_images/static/js/index from ep_copy_paste_images/ep_copy_paste_images
aceInitialized
ep_cursortrace/static/js/main:aceInitialized from ep_cursortrace/cursortrace
postAceInit
ep_scrolltoanchor/static/js/postAceInit:postAceInit from ep_scrolltoanchor/scrolltoanchor
postAceInit
ep_tables2/static/js/datatables:postAceInit from ep_tables2/tables
postAceInit
ep_spellcheck/static/js/spellcheck:postAceInit from ep_spellcheck/ep_spellcheck
postAceInit
ep_mediawiki/static/js/mediawiki from ep_mediawiki/mediawiki
postAceInit
ep_email_notifications/static/js/ep_email:postAceInit from ep_email_notifications/ep_email_notifications
postAceInit
ep_draw/static/js/draw from ep_draw/draw
postAceInit
ep_copy_paste_images/static/js/index from ep_copy_paste_images/ep_copy_paste_images
postAceInit
ep_cursortrace/static/js/main:postAceInit from ep_cursortrace/cursortrace
postAceInit
ep_copy_paste_select_all/static/js/copy_paste_select_all from ep_copy_paste_select_all/copy_paste_select_all
collectContentLineBreak
ep_tables2/static/js/contentcollector:collectContentLineBreak from ep_tables2/tables
collectContentLineText
ep_tables2/static/js/contentcollector:collectContentLineText from ep_tables2/tables
aceKeyEvent
ep_tables2/static/js/datatables:aceKeyEvent from ep_tables2/tables
aceStartLineAndCharForPoint
ep_tables2/static/js/datatables:aceStartLineAndCharForPoint from ep_tables2/tables
aceEndLineAndCharForPoint
ep_tables2/static/js/datatables:aceEndLineAndCharForPoint from ep_tables2/tables
aceAttribsToClasses
ep_tables2/static/js/datatables:aceAttribsToClasses from ep_tables2/tables
aceAttribsToClasses
ep_embedmedia/static/js/hooks:aceAttribsToClasses from ep_embedmedia/embedmedia
aceAttribsToClasses
ep_copy_paste_images/static/js/index from ep_copy_paste_images/ep_copy_paste_images
acePostWriteDomLineHTML
ep_tables2/static/js/datatables:acePostWriteDomLineHTML from ep_tables2/tables
acePostWriteDomLineHTML
ep_authornames/static/js/index from ep_authornames/authornames
acePostWriteDomLineHTML
ep_copy_paste_images/static/js/index from ep_copy_paste_images/ep_copy_paste_images
acePostWriteDomLineHTML
ep_autoscrolldown/static/js/hooks from ep_autoscrolldown/autoscrolldown
disableAuthorColorsForThisLine
ep_tables2/static/js/linestylefilter:disableAuthorColorsForThisLine from ep_tables2/tables
aceEditEvent
ep_tables2/static/js/datatables:aceEditEvent from ep_tables2/tables
aceEditEvent
ep_cursortrace/static/js/main:aceEditEvent from ep_cursortrace/cursortrace
aceEditorCSS
ep_mediawiki/static/js/mediawiki from ep_mediawiki/mediawiki
aceEditorCSS
ep_copy_paste_images/static/js/index from ep_copy_paste_images/ep_copy_paste_images
aceEditorCSS
ep_cursortrace/static/js/css from ep_cursortrace/cursortrace
aceDomLineProcessLineAttributes
ep_copy_paste_images/static/js/index from ep_copy_paste_images/ep_copy_paste_images
collectContentPre
ep_headings/static/js/shared from ep_headings/main
collectContentPre
ep_horizontal_line/static/js/shared from ep_horizontal_line/horizontal_line
collectContentPre
ep_copy_paste_images/static/js/index from ep_copy_paste_images/ep_copy_paste_images
collectContentPre
ep_colors/static/js/shared from ep_colors/main
aceRegisterBlockElements
ep_copy_paste_images/static/js/index from ep_copy_paste_images/ep_copy_paste_images
handleClientMessage_emailSubscriptionSuccess
ep_email_notifications/static/js/ep_email from ep_email_notifications/ep_email_notifications
handleClientMessage_emailUnsubscriptionSuccess
ep_email_notifications/static/js/ep_email from ep_email_notifications/ep_email_notifications
handleClientMessage_emailNotificationGetUserInfo
ep_email_notifications/static/js/ep_email from ep_email_notifications/ep_email_notifications
handleClientMessage_emailNotificationMissingParams
ep_email_notifications/static/js/ep_email from ep_email_notifications/ep_email_notifications
aceInitInnerdocbodyHead
ep_embedmedia/static/js/hooks:aceInitInnerdocbodyHead from ep_embedmedia/embedmedia
aceInitInnerdocbodyHead
ep_cursortrace/static/js/main:aceInitInnerdocbodyHead from ep_cursortrace/cursortrace
aceCreateDomLine
ep_embedmedia/static/js/hooks:aceCreateDomLine from ep_embedmedia/embedmedia
aceCreateDomLine
ep_copy_paste_images/static/js/index:aceCreateDomLine from ep_copy_paste_images/ep_copy_paste_images
aceGetFilterStack
ep_copy_paste_images/static/js/index:aceGetFilterStack from ep_copy_paste_images/ep_copy_paste_images
collectContentPost
ep_copy_paste_images/static/js/index from ep_copy_paste_images/ep_copy_paste_images
collectContentImage
ep_copy_paste_images/static/js/index from ep_copy_paste_images/ep_copy_paste_images
aceAttribClasses
ep_copy_paste_images/static/js/index from ep_copy_paste_images/ep_copy_paste_images
ccRegisterBlockElements
ep_copy_paste_images/static/js/index from ep_copy_paste_images/ep_copy_paste_images
handleClientMessage_CUSTOM
ep_cursortrace/static/js/main from ep_cursortrace/cursortrace
documentReady
ep_cursortrace/static/js/main from ep_cursortrace/cursortrace
documentReady
ep_adminpads/static/js/admin/pads:documentReady from ep_adminpads/ep_adminpad
"

That is the snippet of my settings.json
"
  /\* ep_auth_author allows to add the parameter "authorName" to each here defined user. */
  "ep_auth_author": {
    "prefix": "author/"
  },

/\* Users for basic authentication. is_admin = true gives access to /admin.
     If you do not uncomment this, /admin will not be available! */

  "users": {
    "ldapauth": {
        "url": "ldap://example.org",
        "accountBase": "ou=example,dc=org",
        "accountPattern": "(&(objectClass=_)(sAMAccountName={{username}}))",
        "searchDN": "cn=WebServiceUserAuth,ou=Services,ou=example,dc=org",
        "searchPWD": "password",
        "groupSearchBase": "cn=Etherpad-Admins,ou=Webservices,ou=Groups,ou=example,dc=org",
        "groupAttribute": "member",
        "groupAttributeIsDN": true,
        "searchScope": "sub",
        "groupSearch": "(&(objectClass=_))"
    },
    "author/admin": {
      "password": "password",
      "is_admin": true,
      "authorName": "Administrator"
    },
    "author/guest": {
      "password": "guest",
      "is_admin": false,
      "authorName": "Guest"
    },
    "user": {
      "password": "password",
      "is_admin": false
    },
    "author/username1: {
      "password": "password",
      "is_admin": false,
      "authorName": "Authorname1"
    }
  },
"

I can login with username "author/guest" and password guest but the authorName is not being displayed.

What do I have to check? Which hooks should fire to set the authorName? I am willing to do some debugging and to spend some time fixing this issue. Please give me a little push to the direction I need to go.
Thanks.

Cheers,
Froiloc

**Comment by froiloc** [comment]
I did some debugging and found that in ep_auth_author.json in function exports.handleMessage = function(hook_name, context, cb) the variable context.client.request.session.auth_author is undefined.

There is a context.client.request.session.user though. This one has the settings defined in settings.json, i.e. "password":guest,"is_admin":false,"authorName":Guest,"username":author/guest

I keep looking. maybe I find a fix.

**Comment by froiloc** [comment]
It looks to me like exports.authenticate is not called when the user is logging in. That way initializeAuthAuthorState is not called and therefore context.client.request.session.auth_author is undefined.

**Comment by lsowen** [comment]
@froiloc are you using `author/guest` or `guest` as the username when you log in?

**Comment by froiloc** [comment]
I use author/guest for log in.

I just removed the other plugins that use the authenticate hook. And after that authenticate does get called. Maybe it is a problem in the design of the other plugins. I will continue the checking.

**Comment by lsowen** [comment]
@froiloc Try using just `guest`.  The prefix part (`author/`) is to prevent the default authentication method (which is always called first) from successfully authenticating the user, which would skip calling this plugin.

**TLDR**: Log in without the prefix part of the username.

**Comment by froiloc** [comment]
log in with <code>guest</code> works. <code>exports.authenticate</code> gets called.  <code>context.client.request.session.auth_author</code> is defined. Nevertheless the author name is not set.

I go ahead. I will find the reason. :) Thanks for the help. It is highly appreciated.

**Comment by lsowen** [comment]
@froiloc it might be because you had already logged in with `author/guest`, which is how the user is stored in the backend.   Try adding a new user (eg `author/guest2`) and only logging in with `guest2`.

**Comment by froiloc** [comment]
Nope. This didn't do the trick. The logfile shows

<pre>
ep_auth_author - retrieved authorId null for username author/guest2
...
ep_auth_author - handleMessage called
ep_auth_author - Set author a.qN2b5dXtKGQqzYwa for user author/guest2
ep_auth_author - Set AuthorName to default value of undefined for author a.qN2b5dXtKGQqzYwa
...
</pre>

**Comment by froiloc** [comment]
I tested a bit more.

in ep_auth_author.js
when 
<code>authorManager.setAuthorName(authorId, req.session.auth_author.authorName, function(){</code>
is called, then <code>req.session.auth_author.authorName</code> is not defined.

I replaced it with <code>req.session.user.authorName</code> and **it worked**.

**Comment by froiloc** [comment]
As you have suggested before. The name of a user that has once before logged in has to be set again, because it is stored as "undefined" in the database. But after that change, the name will be loaded when the user logs in again on a later return to the etherpad instance.

I will check out if the whole thing still works after I reinstalled my ep_ldapauth plugin. This is really needed cause it is used for most people her to authenticate.

**Comment by lsowen** [comment]
@froiloc I figured out the issue.  My example config is wrong.  Instead of 

```
{
  "ep_auth_author": {
    "prefix": "author/"
  },
  "users": {
    "author/user": {
      "is_admin": false,
      "authorName": "Test User",
      "password": "changeme1"
    },
    "admin": {
      "is_admin": true,
      "password": "changeme1"
    }
  }
}
```

It should be

```
{
  "ep_auth_author": {
    "prefix": "author/"
  },
  "users": {
    "author/user": {
      "is_admin": false,
      "author_name": "Test User",
      "password": "changeme1"
    },
    "admin": {
      "is_admin": true,
      "password": "changeme1"
    }
  }
}
```

Notice that the author name is set by `author_name` instead of `authorName`.  This allows the user on the front end to set their user name to something different from the dropdown.

**Comment by froiloc** [comment]
Ah, ok. Now I see. :+1: This works. Now I only have to log in with each user I previously used and set the displayed name within the etherpad manually, because else it will use undefined, because the right value is not stored in the database.

Well good that we talked about it. :smile: 

Nevertheless besides this success, this plugin doesn't work with ep_ldapauth because the hook <code>autheticate</code> doesn't get called. I will have to check out why. But this will probably be on another place.

Thank you very much for your support. Good job with the plugin. It is really useful and was one of the top requests here. :) Well done!

Cheers,
Froiloc

**Comment by lsowen** [comment]
@froiloc while more than one authentication plugin can be active at a time, only a single one can successfully authenticate a user.  Therefore, if the authentication for a user is handled by this plugin, no further plugins will be fired.

However, it _should_ be calling the ep_ldapauth plugin if the user is not found by this plugin.
